drivers/counter/counter_mcux_ctimer.c

```diff
@@ -12,7 +12,8 @@
 #include <dt-bindings/clock/mcux_lpc_syscon_clock.h>
 LOG_MODULE_REGISTER(mcux_ctimer, CONFIG_COUNTER_LOG_LEVEL);
 
-#define NUM_CHANNELS 4
+/* One of the CTimer channels is reserved to implement set_top_value API */
+#define NUM_CHANNELS 3
```

Comment:
ctimer_match_t valid value is 0-3. and we use channel 3 to set the top value as match register.
`
typedef enum _ctimer_match
{
    kCTIMER_Match_0 = 0U, /*!< Timer match register 0 */
    kCTIMER_Match_1,      /*!< Timer match register 1 */
    kCTIMER_Match_2,      /*!< Timer match register 2 */
    kCTIMER_Match_3       /*!< Timer match register 3 */
} ctimer_match_t;
`
and we set
' 
struct mcux_lpc_ctimer_data {
	struct mcux_lpc_ctimer_channel_data channels[NUM_CHANNELS];
};
'
maybe we need set a protection in the mcux_lpc_ctimer_set_alarm channel_id